src/main/java/seedu/address/logic/commands/EditTypeCommand.java

```diff
@@ -92,7 +93,8 @@ protected static Event createEditedEvent(Event eventToEdit, EditEventDescriptor
         LocalDate updatedDate = editEventDescriptor.getDate().orElse(eventToEdit.getDate());
         LocalTime updatedTime = editEventDescriptor.getTime().orElse(eventToEdit.getTime());
         Duration updatedDuration = editEventDescriptor.getDuration().orElse(eventToEdit.getDuration());
-
-        return new Event(updatedEventDescription, updatedDate, updatedTime, updatedDuration);
+        RecurFrequency updatedRecurFrequency = editEventDescriptor.getRecurFrequency().orElse(
+                eventToEdit.getRecurFrequency().orElse(null));
```

Comment:
```suggestion
        RecurFrequency updatedRecurFrequency = editEventDescriptor.getRecurFrequency()
                .orElse(eventToEdit.getRecurFrequency().orElse(null));
```
I think the whole method should be together with its parameters in this case